File: QID-1493-MVAdenbank/MVAdenbank.R

# clear variables and close windows
rm(list = ls(all = TRUE))
graphics.off()

# install and load packages
libraries = c("KernSmooth")
lapply(libraries, function(x) if (!(x %in% installed.packages())) {
install.packages(x)
})
lapply(libraries, library, quietly = TRUE, character.only = TRUE)

# load data
x  = read.table("bank2.dat")
x1 = x[1:100, 6]
x2 = x[101:200, 6]

# Compute kernel density estimate
fh1 = bkde(x1, kernel = "biweight")  
fh2 = bkde(x2, kernel = "biweight")  

# plot
plot(fh1, type = "l", lwd = 2, xlab = "Counterfeit                  /                 Genuine", 
    ylab = "Density estimates for diagonals", col = "black", main = "Swiss bank notes", 
    xlim = c(137, 143), ylim = c(0, 0.85)) 
lines(fh2, lty = "dotted", lwd = 2, col = "red3")

File: QID-1493-MVAdenbank/MVAdenbank.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.api as sm
from KDEpy import FFTKDE

# load data
x = pd.read_csv("bank2.dat", sep = "\s+", header=None)
x1 = x.iloc[0:100, 5]
x2 = x.iloc[100:200, 5]

# Compute kernel density estimate
fh1_x, fh1_y = FFTKDE(bw="silverman", kernel='biweight').fit(np.array(x1)).evaluate()
fh2_x, fh2_y = FFTKDE(bw="silverman", kernel='biweight').fit(np.array(x2)).evaluate()

fig, ax = plt.subplots(figsize=(10,10))
ax.plot(fh1_x, fh1_y, c="black")
ax.plot(fh2_x, fh2_y, c="red", ls=":")
plt.xlim(137, 143)
plt.xlabel("Counterfeit                             /                                   Genuine")
plt.ylabel("Density estimates for diagonals")
plt.title("Swiss bank notes")

plt.show()

